### DISCUSSION TITLE
How to use field reference in queries for nested relations.

### DISCUSSION THREAD

**Discussion by The-White-Fang** [prisma/prisma#22639] [Q&A]
### Question

Does [this](https://github.com/prisma/prisma/issues/15068) feature work for a deeply nested relation as well?

Can I do

```
prisma.a.findMany({ where: { field4: x, B: { some: { fieldZ: prisma.A.field3 } } } })
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
model A {
  field1 Int @id
  field2 Int @unique
  field3 Int
  field4 String @db.char(5)
  B      B[]
}

model B {
  fieldX Int @id
  fieldY Int
  fieldZ Int

  rel A @relation(fields: [fieldY], references: [field2])
}
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Debian
- Database: MySQL
- Node.js version: 16.19.0

- Run `prisma -v` to see your Prisma version and paste it
prisma                  : 5.4.2
@prisma/client          : 5.4.2
Current platform        : debian-openssl-3.0.x
Query Engine (Node-API) : libquery-engine ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules/@prisma/engines/libquery_engine-debian-openssl-3.0.x.so.node)
Schema Engine           : schema-engine-cli ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules/@prisma/engines/schema-engine-debian-openssl-3.0.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.4.1-2.ac9d7041ed77bcc8a8dbd2ab6616b39013829574
Default Engines Hash    : ac9d7041ed77bcc8a8dbd2ab6616b39013829574
Studio                  : 0.494.0

**Comment by ludralph** [comment]
Hi @The-White-Fang 👋 

Thank you for raising this question.

The query works with fieldReferences.

```js
prisma.a.findMany({ where: { field4: x, B: { some: { fieldZ: prisma.A.field3 } } } })
```

A sample create using the model you shared is shown below:
```js
const res = await prisma.a.create({
  data:{
    field1:1,
    field2:2,
    field3:3,
    field4: 'test',
    B:{
      create:{
        fieldX: 1,
        fieldZ: 3,
      }
    }
  }
})
```

The output is:
```
[ { field1: 1, field2: 2, field3: 3, field4: 'test' } ]
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by The-White-Fang** [reply]
Thanks. I understand what I was doing wrong. Seems like the types are not generated correctly for these, which is why I was using `prisma.a.fields.field3`. Is this a known bug? Or are there any options in the `PrismaClient` constructor that are required to be passed to correctly get the type information?

**Reply by ludralph** [reply]
Hi @The-White-Fang 

I'm not aware of any bug specifically related to type generation for field references, nor are there special PrismaClient constructor options required for this feature.

**Reply by The-White-Fang** [reply]
Hi @ludralph 

I am unsure if this works in the latest version, but in v5.4.2 it didn't. I will test it in the newest version again, if the types are working an update here.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
